### PR TITLE
Updated docs/setup for using docker for a local database

### PR DESCRIPTION
**Pull request by ktuite**
This is a follow-up to https://github.com/getodk/central-backend/pull/396. That PR lets you use docker to run a local postgres instance, but it uses the CircleCI database init script that only creates a test database. So it works for testing, but not for running a dev server. 

Now `make run-docker-postgres` will make both test and dev databases.

I decided to also add some of the CLI user creation commands to the backend README here, because that's where I've hoped to find them. Now someone with a fresh database can get into using Central right away without hunting all through the docs, but they can dig into the rest of the docs if they wish.

#### What has been done to verify that this works as intended?

Works on my machine? If anyone else wants to try it, please do! 

#### Why is this the best possible solution? Were any other approaches considered?

Not sure. Maybe a separate CircleCI script that is not actually used by CircleCI is not ideal but I couldn't think of anything better.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Only used for developers - probably makes things easier and more reproducible.

#### Does this change require updates to the API documentation? If so, please update docs/api.md as part of this PR.

No,  just the readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,30 +38,30 @@ To run the server, run `make run` from the repository root. Once started, the se
...
 
 With Central running (e.g. `make run` in another terminal window) use the command line interface to create a user and promote them to an admin role. The same commands can be used to create additional users.
 
-`node lib/bin/cli.js -u <email> user-create` (will prompt for password)
-`node lib/bin/cli.js -u <email> user-promote`
-
+```
+node lib/bin/cli.js -u SOME_EMAIL user-create # will prompt for password
+node lib/bin/cli.js -u SOME_EMAIL user-promote
+```
```
Comment (matthew-white) [inline README.md:64]: These two commands were rendering on a single line, so I put them in a code block. I also changed `<email>` to `SOME_EMAIL` with the thought that someone might actually enter the `<` and `>`. 😆

**Review by matthew-white — COMMENTED** [review]
I'm pretty excited about this PR! I've been having trouble with some tests for a while (I think ones using `testServiceFullTrx()` or `testContainerFullTrx()`), and I'm hoping that this PR will resolve those. @ktuite mentioned that she was also having trouble with some tests, and it sounds like we may be seeing the same thing. @ktuite says that by using Postgres in Docker, which this PR facilitates, those issues went away for her.

Specifically, some tests have been timing out for me. Actually, with the change in #558, those tests now cause the entire test suite to fail fast. I see the following output:

```
      1) "after each" hook for "should reject keyless forms in keyed projects @slow"


  893 passing (1m)
  2 pending
  1 failing

  1) "after each" hook for "should reject keyless forms in keyed projects @slow":
     error: drop owned by current_user - cache lookup failed for function 422805
      at Parser.parseErrorMessage (node_modules/pg-protocol/dist/parser.js:278:15)
      at Parser.handlePacket (node_modules/pg-protocol/dist/parser.js:126:29)
      at Parser.parse (node_modules/pg-protocol/dist/parser.js:39:38)
      at Socket.<anonymous> (node_modules/pg-protocol/dist/index.js:10:42)
      at Socket.emit (domain.js:470:12)
      at addChunk (internal/streams/readable.js:290:12)
      at readableAddChunk (internal/streams/readable.js:265:9)
      at Socket.Readable.push (internal/streams/readable.js:204:10)
      at TCP.onStreamRead (internal/stream_base_commons.js:188:23)
```

I'm not quite sure how my database got in this state. I wonder if it could be related to the Postgres version. I thought I was running 9.6, but apparently I'm running 13.2. 😅 One thing that's nice about this PR is that it makes it easier for everyone to run the same version of Postgres.

I made some changes to the PR after discussing with @ktuite. The main change I made was to have there be a single script to create databases in a Docker environment (for either CircleCI or `make run-docker-postgres`). The script, `create-docker-databases.js`, will create both `jubilant` and `jubilant_test`. We don't really need `jubilant` in testing, but I think it's OK to always create it. (The script runs in less than a second in CircleCI.)

My commit is a work-in-progress, so I've marked this PR as a draft, and I may force-push changes later. But I wanted to push what I have so far to give a sense of where the PR is heading.

**Comment by matthew-white** [comment]
@ktuite and I chatted about this again today. The next step is that @ktuite is going to try out this branch locally.

**Review by matthew-white — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -38,23 +38,31 @@ To run the server, run `make run` from the repository root. Once started, the se
...
 
-
 ### Setting up the database manually
 
 First, create a database and user in Postgres. Either use the same settings as the [default configuration file](config/default.json), or update your local configuration file to match the settings you choose. For example:
 
 ```sql
 CREATE USER jubilant WITH PASSWORD 'jubilant';
-CREATE DATABASE jubilant_test with owner=jubilant encoding=UTF8;
+CREATE DATABASE jubilant_test WITH OWNER=jubilant ENCODING=UTF8;
```
[the remark this answers is not in the document]
Comment (alxndrsn) [reply README.md:47]: @matthew-white is that more consistent?  Surely that would be:
```
CREATE DATABASE JUBILANT_TEST WITH OWNER=JUBILANT ENCODING=UTF8;
```
:stuck_out_tongue_closed_eyes: 

Can I vote for capitalised keywords, but lower-case table and column names?  i.e.

```
CREATE DATABASE jubilant_test WITH owner=jubilant encoding=UTF8;
```

**Comment by alxndrsn** [comment]
@matthew-white it looks like your test error at https://github.com/getodk/central-backend/pull/577#pullrequestreview-1075038314 is the result of an open stream connection to postgres.  Does increasing the test timeout get rid of this error?

**Pull request merged**